triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRuleSimulationLibTest.scala

```diff
@@ -66,142 +79,482 @@ class TriggerRuleSimulationLibTest
-          startState = converter
+      for {
+        client <- ledgerClient()
+        party <- allocateParty(client)
+        (trigger, simulator) = getSimulator(
+          client,
+          QualifiedName.assertFromString("Cats:feedingTrigger"),
+          packageId,
+          applicationId,
+          compiledPackages,
+          config.participants(ParticipantId).apiServer.timeProviderType,
+          triggerRunnerConfiguration,
+          party,
+        )
+        converter = new Converter(compiledPackages, trigger)
+        result <- forAll(updateState) { case (acs, userState, inFlightCmds, msg) =>
+          val startState = converter
             .fromTriggerUpdateState(
               acs,
               userState,
-              Map.empty,
+              inFlightCmds,
               TriggerParties(Party(party), Set.empty),
               triggerRunnerConfiguration,
             )
-          (submissions, metrics, endState) <- simulator.updateStateLambda(startState, msg)
-        } yield {
-          metrics.evaluation.submissions should be(submissions.size)
-          metrics.submission.creates should be(submissions.map(numberOfCreateCommands).sum)
-          metrics.submission.createAndExercises should be(
-            submissions.map(numberOfCreateAndExerciseCommands).sum
-          )
-          metrics.submission.exercises should be(submissions.map(numberOfExerciseCommands).sum)
-          metrics.submission.exerciseByKeys should be(
-            submissions.map(numberOfExerciseByKeyCommands).sum
-          )
-          metrics.evaluation.steps should be(
-            metrics.evaluation.getTimes + metrics.evaluation.submissions + 1
-          )
-          Some(metrics.startState.acs.activeContracts) should be(
-            numberOfActiveContracts(startState, trigger.level, trigger.version)
-          )
-          Some(metrics.startState.acs.pendingContracts) should be(
-            numberOfPendingContracts(startState, trigger.level, trigger.version)
-          )
-          Some(metrics.startState.inFlight.commands) should be(
-            numberOfInFlightCommands(startState, trigger.level, trigger.version)
-          )
-          Some(metrics.endState.acs.activeContracts) should be(
-            numberOfActiveContracts(endState, trigger.level, trigger.version)
-          )
-          Some(metrics.endState.acs.pendingContracts) should be(
-            numberOfPendingContracts(endState, trigger.level, trigger.version)
-          )
-          Some(metrics.endState.inFlight.commands) should be(
-            numberOfInFlightCommands(endState, trigger.level, trigger.version)
-          )
 
-          val pendingContractSubmissions = submissions
-            .map(request =>
-              numberOfCreateCommands(request) + numberOfCreateAndExerciseCommands(request)
-            )
-            .sum
+          for {
+            (submissions, metrics, endState) <- simulator.updateStateLambda(startState, msg)
+          } yield {
```

Comment:
Ad hoc timing data suggested that this style of test setup would be more efficient